tests/frontend/test_program.py

```diff
@@ -669,5 +620,6 @@ def test_GBS_success(self):
 
         prog = prog.compile('gbs')
         assert len(prog) == 4
-        assert prog.circuit[-1].op.__class__ == ops.MeasureFock
-        assert [x.ind for x in prog.circuit[-1].reg] == list(range(3))
+        last_cmd = prog.circuit[-1]
+        assert last_cmd.op.__class__ == ops.MeasureFock
+        assert [x.ind for x in last_cmd.reg] == list(range(3))
```

Comment:
@smite any tests you can think of that test interaction of `compile` and `optimize`?

Reply:
or tests that test interaction between `p=prog.compile` and subsequent `eng.run(p)` (which has implicity compile to backend)?

Reply:
Not right now, but we have already some tests which first compile and then run the program.